### PR TITLE
Fix formatting issues and styling. Anchor elements to redirect to new tab.

### PR DESCRIPTION
**Pull request by MikolJiron**

### CODE REVIEW

**Review by Lorraine-Lyu — APPROVED** [review]

File: src/main/webapp/js/datahandlers/collegelistdatahandler.js [hunk 1]
```diff
@@ -52,7 +54,7 @@ class CollegeListDataHandler extends ListDataHandler {
       element[ID].toString(),
       element[NAME],
       element[ACCEPTANCE_RATE].toString(),
-      element[ACT_SCORE].toString(),
+      DOLLAR_SIGN.concat(integerWithCommas(element[ANNUAL_COST])),
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/collegelistdatahandler.js]: Make another util method to create this dollar value.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by tinagrasso — APPROVED** [review]

File: src/main/webapp/js/datahandlers/utils.js [hunk 2]
```diff
@@ -26,4 +28,22 @@ const addSpaceToCamelCase = (str) => {
...
 
-exports = {addSpaceToCamelCase};
\ No newline at end of file
+/**
+ * This method returns a string representing a number with commas.
+ * @param {number} num - The number to convert.
+ * @returns {!string} - The string representation of the number with commas.
+ */
+const integerWithCommas = (num) => {
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
```
Comment (hzimz) [inline src/main/webapp/js/datahandlers/utils.js]: Move this regex into a constant that describes the purpose of this regex.

**Review by hzimz — CHANGES_REQUESTED** [review]

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Can you rebase devel so we can test it for warnings?

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Some linter messages:
```
src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'ACT_SCORE' is never referenced in this file
  22| const {ACCEPTANCE_RATE, ACT_SCORE, CollegeQueryBuilder, ID, NAME} = goog.require('datahandlers.collegequerybuilder');
                              ^^^^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:61: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  61|     return {
          ^^^^^^^^
  62|       schoolName : element[NAME],
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  77|           convertToDollar(element[MEDIAN_DEBT])
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  78|     }
      ^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:85: WARNING - [JSC_UNUSED_LOCAL_ASSIGNMENT] Value assigned to local variable loadErrorPage_ is never read
  85| const loadErrorPage_ = (element, occurrence, action, error) => {
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  86|   // TODO: This is being rewritten.
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  87| };
      ^

src/main/webapp/js/datahandlers/collegequerybuilder.js:36: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  36| const NAME =  'school.name'
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/utils.js:31: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  31| const addSpaceToCamelCase = (str) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  32|   let result = str.replace( /([A-Z])/g, " $1" );
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  33|   return result.charAt(0).toUpperCase() + result.slice(1);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  34| }
      ^

src/main/webapp/js/datahandlers/utils.js:39: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  39|  * @returns {!string} - The string representation of the number with commas.
                   ^

src/main/webapp/js/datahandlers/utils.js:41: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  41| const integerWithCommas = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  42|   return num.toString().replace(NUM_REGEX, ',');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  43| }
      ^

src/main/webapp/js/datahandlers/utils.js:48: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  48|  * @returns {!string} - The string representation of the number with commas and $.
                   ^

src/main/webapp/js/datahandlers/utils.js:50: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  50| const convertToDollar = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  51|   return DOLLAR_SIGN.concat(integerWithCommas(num));
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  52| }
      ^

src/main/webapp/js/datahandlers/utils.js:29: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  29|  * @returns The formatted string.
         ^
```

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Some linter messages:
```
src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'ACT_SCORE' is never referenced in this file
  22| const {ACCEPTANCE_RATE, ACT_SCORE, CollegeQueryBuilder, ID, NAME} = goog.require('datahandlers.collegequerybuilder');
                              ^^^^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:61: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  61|     return {
          ^^^^^^^^
  62|       schoolName : element[NAME],
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  77|           convertToDollar(element[MEDIAN_DEBT])
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  78|     }
      ^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:85: WARNING - [JSC_UNUSED_LOCAL_ASSIGNMENT] Value assigned to local variable loadErrorPage_ is never read
  85| const loadErrorPage_ = (element, occurrence, action, error) => {
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  86|   // TODO: This is being rewritten.
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  87| };
      ^

src/main/webapp/js/datahandlers/collegequerybuilder.js:36: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  36| const NAME =  'school.name'
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/utils.js:31: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  31| const addSpaceToCamelCase = (str) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  32|   let result = str.replace( /([A-Z])/g, " $1" );
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  33|   return result.charAt(0).toUpperCase() + result.slice(1);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  34| }
      ^

src/main/webapp/js/datahandlers/utils.js:39: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  39|  * @returns {!string} - The string representation of the number with commas.
                   ^

src/main/webapp/js/datahandlers/utils.js:41: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  41| const integerWithCommas = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  42|   return num.toString().replace(NUM_REGEX, ',');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  43| }
      ^

src/main/webapp/js/datahandlers/utils.js:48: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  48|  * @returns {!string} - The string representation of the number with commas and $.
                   ^

src/main/webapp/js/datahandlers/utils.js:50: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  50| const convertToDollar = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  51|   return DOLLAR_SIGN.concat(integerWithCommas(num));
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  52| }
      ^

src/main/webapp/js/datahandlers/utils.js:29: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  29|  * @returns The formatted string.
         ^
```More linter messages:
```
src/main/webapp/js/datahandlers/collegepagedatahandler.js:22: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const {ACCEPTANCE_RATE, ACT_SCORE, ANNUAL_COST, CollegeQueryBuilder, FIFTH_NET_COST, FIRST_NET_COST, FOURTH_NET_COST, MEDIAN_DEBT, NAME, SECOND_NET_COST, THIRD_NET_COST} = goog.require('datahandlers.collegequerybuilder');
const {SinglePageDataHandler} = goog.require('datahandlers.singlepagedatahandler');
const {convertToDollar} = goog.require('datahandlers.utils');

  22| const {ACCEPTANCE_RATE, ACT_SCORE, CollegeQueryBuilder, ANNUAL_COST, FIFTH_NET_COST, FIRST_NET_COST, FOURTH_NET_COST, MEDIAN_DEBT, NAME, SECOND_NET_COST, THIRD_NET_COST} = goog.require('datahandlers.collegequerybuilder');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:61: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  61|     return {
          ^^^^^^^^
  62|       schoolName : element[NAME],
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  71|       cumulativeMedianDebt : convertToDollar(element[MEDIAN_DEBT])
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  72|     }
      ^^^^^
```

**Comment by MikolJiron** [comment]
@CarlosM10011 I don't think you ran the linter on the latest version of college-link branch. Did you pull?

**Comment by MikolJiron** [comment]
@CarlosM10011 Please pull before running the linter, it seems some of the linter messages you included in your last comment had already been resolved in one of my previous commits.

**Review by CarlosM10011 — APPROVED** [review]
Some linter messages:
```
src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'ACT_SCORE' is never referenced in this file
  22| const {ACCEPTANCE_RATE, ACT_SCORE, CollegeQueryBuilder, ID, NAME} = goog.require('datahandlers.collegequerybuilder');
                              ^^^^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:61: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  61|     return {
          ^^^^^^^^
  62|       schoolName : element[NAME],
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  77|           convertToDollar(element[MEDIAN_DEBT])
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  78|     }
      ^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:85: WARNING - [JSC_UNUSED_LOCAL_ASSIGNMENT] Value assigned to local variable loadErrorPage_ is never read
  85| const loadErrorPage_ = (element, occurrence, action, error) => {
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  86|   // TODO: This is being rewritten.
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  87| };
      ^

src/main/webapp/js/datahandlers/collegequerybuilder.js:36: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  36| const NAME =  'school.name'
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/utils.js:31: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  31| const addSpaceToCamelCase = (str) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  32|   let result = str.replace( /([A-Z])/g, " $1" );
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  33|   return result.charAt(0).toUpperCase() + result.slice(1);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  34| }
      ^

src/main/webapp/js/datahandlers/utils.js:39: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  39|  * @returns {!string} - The string representation of the number with commas.
                   ^

src/main/webapp/js/datahandlers/utils.js:41: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  41| const integerWithCommas = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  42|   return num.toString().replace(NUM_REGEX, ',');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  43| }
      ^

src/main/webapp/js/datahandlers/utils.js:48: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] string is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  48|  * @returns {!string} - The string representation of the number with commas and $.
                   ^

src/main/webapp/js/datahandlers/utils.js:50: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  50| const convertToDollar = (num) => {
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  51|   return DOLLAR_SIGN.concat(integerWithCommas(num));
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  52| }
      ^

src/main/webapp/js/datahandlers/utils.js:29: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  29|  * @returns The formatted string.
         ^
```More linter messages:
```
src/main/webapp/js/datahandlers/collegepagedatahandler.js:22: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const {ACCEPTANCE_RATE, ACT_SCORE, ANNUAL_COST, CollegeQueryBuilder, FIFTH_NET_COST, FIRST_NET_COST, FOURTH_NET_COST, MEDIAN_DEBT, NAME, SECOND_NET_COST, THIRD_NET_COST} = goog.require('datahandlers.collegequerybuilder');
const {SinglePageDataHandler} = goog.require('datahandlers.singlepagedatahandler');
const {convertToDollar} = goog.require('datahandlers.utils');

  22| const {ACCEPTANCE_RATE, ACT_SCORE, CollegeQueryBuilder, ANNUAL_COST, FIFTH_NET_COST, FIRST_NET_COST, FOURTH_NET_COST, MEDIAN_DEBT, NAME, SECOND_NET_COST, THIRD_NET_COST} = goog.require('datahandlers.collegequerybuilder');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:61: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  61|     return {
          ^^^^^^^^
  62|       schoolName : element[NAME],
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  71|       cumulativeMedianDebt : convertToDollar(element[MEDIAN_DEBT])
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  72|     }
      ^^^^^
```

**Comment by CarlosM10011** [comment]
My mistake. I guess I pulled and ran against the wrong branch? They seem to be gone now :)

**Review by hzimz — APPROVED** [review]

**Pull request merged**